languages/go/exercises/concept/numbers-floating-point/numbers-floating-point_test.go

```diff
@@ -0,0 +1,200 @@
+package numbers_floating_point
+
+import (
+	"testing"
+)
+
+func TestAnnualBalanceUpdate(t *testing.T) {
+	tests := []struct {
+		name    string
+		balance float64
+		want    float64
+	}{
+		{
+			name:    "AnnualBalanceUpdateForEmptyStartBalance",
+			balance: 0,
+			want:    0,
+		},
+		{
+			name:    "AnnualBalanceUpdateForSmallPositiveStartBalance",
+			balance: 0.000001,
+			want:    0.000001005,
+		},
+		{
+			name:    "AnnualBalanceUpdateForAveragePositiveStartBalance",
+			balance: 1000,
+			want:    1016.21,
+		},
+		{ //test case fails in Golang with float64
```

Comment:
This test case from C# fails in Go with a float64

Reply:
floats can not be tested with `==` or `!=`. They need to be tested as:

```go
eps := 0.01
if eps < math.Abs(got - tt.want) {
    t.Errorf(...)
}
```

Reply:
where eps (epsilon) is the acceptable tolerance